File: dino-reporting/dino-core/src/main/scala/com/walmart/xtools/dino/core/functesting/EventParseUtil.scala
package com.walmart.xtools.dino.core.functesting

import com.github.wnameless.json.flattener.JsonFlattener

import com.google.gson.Gson

import com.lucidworks.spark.util.SolrSupport

import org.apache.solr.common.SolrInputDocument

import java.text.ParseException

import EventParseUtil._

object EventParseUtil {

  def convertData(incomingRecord: String): SolrInputDocument = {
    val jsonStr: String = JsonFlattener.flatten(incomingRecord)

    val gson: Gson = new Gson()

    val eventDataElement: ResultData =
      gson.fromJson(jsonStr, classOf[ResultData])

    val solrDoc: SolrInputDocument = SolrSupport.autoMapToSolrInputDoc(
      String.valueOf(1234),
      eventDataElement,
      null)
    //eventDataElement.getResultData().getId()
    solrDoc
  }

}

class EventParseUtil

File: dino-reporting/dino-core/src/main/java/com/walmart/xtools/dino/core/utils/Solr/EventParseUtil.java
package com.walmart.xtools.dino.core.utils.Solr;

import com.github.wnameless.json.flattener.JsonFlattener;
import com.google.gson.Gson;
import com.lucidworks.spark.util.SolrSupport;
import org.apache.solr.common.SolrInputDocument;

import java.text.ParseException;



public class EventParseUtil {

    /*public EventParseUtil() {
        super();
    }


    static SolrInputDocument convertData (String incomingRecord) throws NumberFormatException, ParseException {

        String jsonStr = JsonFlattener.flatten(incomingRecord);
        //System.out.println("Here is the flatten json "+ jsonStr);
        Gson gson = new Gson();
        ResultData eventDataElement = gson.fromJson(jsonStr,ResultData.class);

        SolrInputDocument solrDoc =
                SolrSupport.autoMapToSolrInputDoc(String.valueOf(1234), eventDataElement, null);
        //eventDataElement.getResultData().getId()
        return solrDoc;
    }*/
}
